batch10/assignment2/assignment2-hassan-shaikh.md

```diff
@@ -0,0 +1,28 @@
+Q1) Share screenshot of the above Branch Protection of `main` branch
+Ans: Done and uploaded on the assignement 2 at Dice LMS Portal
+
+Q2) Explain Docker Containers vs VMs
+Ans: 
+- VM has Hardware Level Virtualization where as Containers have OS Level Virtualization
+- VM share hardware but has own OS, where as Container share hardware as well as OS Kernel, but it has its own OS. 
+- VM takes more space as compare to Containers.
+
+Q3) Explain Docker Architecture
+Ans: In Docker, each container shares the same OS Kernel.Kubernettes is used to do clustering. 
```

Comment:
Question was for Docker Architecture...
Meaning client(docker cli), server(docker daemon), registry, etc